src/jog/jog.service.ts

```diff
@@ -0,0 +1,54 @@
+import {BadRequestException, Injectable, NotFoundException} from '@nestjs/common';
+import {InjectModel} from "@nestjs/mongoose";
+import * as mongoose from "mongoose";
+import {Jog} from "./schemas/jog.schema";
+import {Query} from 'express-serve-static-core';
+import { User } from 'src/auth/schemas/user.schema';
+
+
+@Injectable()
+export class JogService {
+    constructor(
+        @InjectModel(Jog.name)
+        private jogModel: mongoose.Model<Jog>
+    ) {}
+
+    async findAll(query: Query): Promise<Jog[]> {
+        const resultsPerPage = 2;
+        const currentPage = Number(query.page) || 1;
+        const skip = resultsPerPage * (currentPage - 1);
+
+        const jogs = await this.jogModel.find().limit(resultsPerPage).skip(skip);
+        return jogs;
+    }
+
+    async create(jog: Jog, user: User): Promise<Jog> {
+        
+        const data = Object.assign(jog,{ user: user._id})
+
+        const response = await this.jogModel.create(jog);
+        return response;
+    }
+
+    async findById(id: string): Promise<Jog> {
+
+        const jog = await this.jogModel.findById(id);
+
+        if(!jog) {
```

Comment:
what if a user tries to get a resource for another user? shouldn't we also prevent that?

Reply:
absolutely, fixing in the upcoming push🙏